### PR TITLE
upgrade library manager to download subdependencies too

### PR DESCRIPTION
**Pull request by PranavPurwar**
Also improves the performance and experience significantly

### CODE REVIEW

File: app/src/main/java/dev/aldi/sayuti/editor/manage/ManageLocalLibraryActivity.java [hunk 1]
```diff
@@ -54,23 +61,84 @@ private void initToolbar() {
...
+                return;
+            }
+            var group = parts[0];
+            var artifact = parts[1];
+            var version = parts[2];
+            var resolver = new DependencyResolver(group, artifact, version);
+            Executors.newSingleThreadExecutor().execute(() -> resolver.resolveDependency(new DependencyResolver.DependencyResolverCallback() {
+                @Override
+                public void invalidPackaging(@NonNull String dep) {
+                    new Handler(Looper.getMainLooper()).post(() -> text.setText("Invalid packaging for dependency " + dep));
```
Comment (JavkhlanK) [inline app/src/main/java/dev/aldi/sayuti/editor/manage/ManageLocalLibraryActivity.java:91]: early review: please create just 1 `Handler` instance

**Comment by JavkhlanK** [comment]
Looking good so far. Just one issue, the entire app crashes thanks to an uncaught FileNotFoundException when trying to download the artifact `com.appodeal.ads.sdk.networks:appodeal:3.0.2.0`.

**Comment by PranavPurwar** [comment]
> Looking good so far. Just one issue, the entire app crashes thanks to an uncaught FileNotFoundException when trying to download the artifact `com.appodeal.ads.sdk.networks:appodeal:3.0.2.0`.

Fixed

File: app/build.gradle [hunk 2]
```diff
@@ -98,17 +98,17 @@ dependencies {
...
+    implementation "com.google.android.material:material:1.9.0"
     implementation "com.airbnb.android:lottie:3.1.0"
     implementation "com.google.android.flexbox:flexbox:3.0.0"
 
-    implementation "androidx.annotation:annotation:1.5.0"
-    implementation "com.android.tools:annotations:30.4.1"
-    implementation "com.google.guava:guava:31.1-jre"
+    implementation "androidx.annotation:annotation:1.6.0"
+    implementation "com.android.tools:annotations:31.0.2"
+    implementation "com.google.guava:guava:32.0.1-jre"
```
Comment (JavkhlanK) [inline app/build.gradle:111]: cmon at least make a separate commit for this

<sub><sup>not saying you should fix that rn, just don't in the future</sup></sub>

Comment (PranavPurwar) [reply app/build.gradle:111]: okay

**Comment by hasrat-ali** [comment]
https://github.com/Sketchware-Pro/Sketchware-Pro/assets/42956453/014acc3e-82a7-4fd3-a6ee-a166bd98fd65

Nice 👍👍👍

**Comment by JavkhlanK** [comment]
1. https://stackoverflow.com/a/54985015
2. you also wrote `matetial` instead of `material`

**Comment by hasrat-ali** [comment]
Ohh yeah, but still not downloading

**Comment by PranavPurwar** [comment]
> Ohh yeah, but still not downloading

Check the solutions provided in the stackoverflow post

**Pull request merged**